src/components/draw/DrawDirective.js

```diff
@@ -27,889 +28,798 @@ goog.require('ga_permalink');
    *                        automatically on s3.
    *
    */
-  module.directive('gaDraw',
-    function($timeout, $translate, $window, $rootScope, gaBrowserSniffer,
-        gaDefinePropertiesForLayer, gaDebounce, gaFileStorage, gaLayerFilters,
-        gaExportKml, gaMapUtils, gaPermalink, gaUrlUtils,
-        $document, gaMeasure) {
-
-      var createDefaultLayer = function(map, useTemporaryLayer) {
-        // #2820: we set useSpatialIndex to false to allow display of azimuth
-        // circle created by style.
-        var dfltLayer = new ol.layer.Vector({
-          source: new ol.source.Vector({useSpatialIndex: false}),
-          visible: true
-        });
-        gaDefinePropertiesForLayer(dfltLayer);
-
-        if (useTemporaryLayer) {
-          dfltLayer.displayInLayerManager = false;
-          dfltLayer.preview = true;
+  module.directive('gaDraw', function($translate, $rootScope, $timeout,
+      gaBrowserSniffer, gaDefinePropertiesForLayer, gaDebounce, gaFileStorage,
+      gaLayerFilters, gaExportKml, gaMapUtils, $document, gaMeasure,
+      gaStyleFactory) {
+
+    var createDefaultLayer = function(map, useTemporaryLayer) {
+      // #2820: we set useSpatialIndex to false to allow display of azimuth
+      // circle created by style.
+      var dfltLayer = new ol.layer.Vector({
+        source: new ol.source.Vector({useSpatialIndex: false}),
+        visible: true
+      });
+      gaDefinePropertiesForLayer(dfltLayer);
+
+      if (useTemporaryLayer) {
+        dfltLayer.displayInLayerManager = false;
+        dfltLayer.preview = true;
+      }
+      gaMeasure.registerOverlaysEvents(map, dfltLayer);
+      dfltLayer.label = $translate.instant('draw_layer_label');
+      dfltLayer.type = 'KML';
+      return dfltLayer;
+    };
+
+    // Creates a new help tooltip
+    var createHelpTooltip = function() {
+      var tooltipElement = $document[0].createElement('div');
+      tooltipElement.className = 'ga-draw-help';
+      var tooltip = new ol.Overlay({
+        element: tooltipElement,
+        offset: [15, 15],
+        positioning: 'top-left',
+        stopEvent: false
+      });
+      return tooltip;
+    };
```

Comment:
Looks to me like this div is always visible, even when empty. Could we avoid that?